torchvtk/augmentation/DictTransform.py

```diff
@@ -0,0 +1,150 @@
+import math
+import random
+from abc import abstractmethod
+
+import numpy as np
+import torch
+from torch.nn import functional as f
+
+
+class DictTransform(object):
+
+    def __init__(self, device="cpu", apply_on=["vol", "mask"], dtype=torch.float32, batch_transform=False):
+        super().__init__()
+
+        self.device = device
+        self.apply_on = apply_on
+        self.dtype = dtype
+        self.batch_transform = batch_transform
+
+    @abstractmethod
+    def transform(self, data): pass
+
+    def __call__(self, data):
+
+        for key in self.apply_on:
+            tmp = data[key]
+
+            if not torch.is_tensor(tmp):
+                tmp = torch.from_numpy(tmp)
+
+            # to GPU.
+            if self.device == "cuda":
+                tmp = tmp.to(self.device)
+
+            if self.dtype is torch.float16:
+                tmp = tmp.to(torch.float32)
+
+            if self.batch_transform is False:
+                tmp = self.transform(tmp)
+            else:
+                batch = []
+                for x in range(tmp.shape[0]):
+                    # get batch thing
+                    tmptensor = tmp[x, ...]
+                    # transform
+                    batch.append(self.transform(tmptensor))
+                # stack tensor back
+                tmp = torch.stack(batch, dim=0)
```

Comment:
`batch_transform` should work the other way right? If its true, put in the whole thing.
I'm still not sure if this is the best place to take care of that. 

Maybe this `batch_transform` flag should rather be on the class level and should be hidden from the actual API? I mean each transform should be defined so it works with both 4D tensors and 5D tensors. This flag should then only split the batch if for example a rotation with different angles can't be easily parallelized over the whole batch. The user however should never have to decice wether a transform is applied on a batch or single item. The implementation of the transforms should take care of this. 

I'm suggesting we keep this batch_transform flag, but set it fixed in each transform's `__init__`.

Reply:
I agree that it is not very elegant. i removed it completely and changed the code so that it does not matter. it is tested with 4D and 5D tensor. Not tested with 2D or 3D tho. But that should not be a problem, because its kinda out of scope or?